### PR TITLE
[NT-593] Editorial Project Navigation

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Fills in the navigation from `DiscoveryEditorialCell` -> `DiscoverPageViewController` -> `EditorialProjectsViewController` -> `DiscoverPageViewController` -> `ProjectNavigatorViewController` and ensures that the correct `RefTag` makes it through to the end.

# 🤔 Why

Continuation of the works for our Go Rewardless editorial campaign.

# 🛠 How

- Updated delegate methods to pass `DiscoveryParams.TagID` value.
- Inferred the `RefTag` from `DiscoveryParams` as it had been done before.
- Updated tests.

# ✅ Acceptance criteria

- [x] Tap on the Go Rewardless Editorial cell on Discovery, select a project from the list, using a breakpoint or Charles, verify that you're seeing the correct `RefTag`.

### CODE REVIEW

File: Library/ViewModels/DiscoveryPageViewModel.swift [hunk 1]
```diff
@@ -424,6 +424,10 @@ private func saveSeen(activities: [Activity]) {
 }
 
 private func refTag(fromParams params: DiscoveryParams, project _: Project) -> RefTag {
+  if params.tagId != nil, params.tagId == .goRewardless {
```
Comment (justinswart) [inline Library/ViewModels/DiscoveryPageViewModel.swift]: It seems `RefTag`s are inferred in this function. I realize that `Project` is not needed but I am still trying to change as little as possible 😅

Comment (ifbarrera) [reply Library/ViewModels/DiscoveryPageViewModel.swift]: Any reason you've chosen to hard-code the `tagId` type here? Couldn't you just pass in the `tagId` to the `.projectCollection` refTag?

```
if let tagId = params.tagId {
   return .projectCollection(tagId)
}
```

File: Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift [hunk 2]
```diff
@@ -210,8 +210,8 @@ internal final class DiscoveryPageViewController: UITableViewController {
 
     self.viewModel.outputs.goToEditorialProjectList
       .observeForControllerAction()
-      .observeValues { [weak self] tag, refTag in
-        self?.goToEditorialProjectList(using: tag, refTag: refTag)
+      .observeValues { [weak self] tagId in
+        self?.goToEditorialProjectList(using: tagId)
```
Comment (ifbarrera) [inline Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift]: We have a tracking event for tapping the editorial header that'll need the `refTag` passed in 🤔 Do you think we could just generate it dynamically by doing something like:

```
 AppEnvironment.current.koala.trackEditorialHeaderTapped(refTag: RefTag.projectCollection(.goRewardless))
```

Comment (justinswart) [reply Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift]: Yeah I think that should be fine?

**Pull request merged**